core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp

```diff
@@ -534,7 +534,8 @@ struct CudaReductionsFunctor<FunctorType, ArgTag, false, true> {
     __syncthreads();
     unsigned int num_teams_done = 0;
     if (threadIdx.x + threadIdx.y == 0) {
-      num_teams_done = Kokkos::atomic_fetch_add(global_flags, 1) + 1;
+      num_teams_done =
+          Kokkos::atomic_fetch_add(global_flags, (unsigned int)1) + 1;
```

Comment:
Did it not compile w/o casting?  Also probably deduce the type pointed to by `global_flags` if it is really necessary.

Reply:
it did compile but gave tons of warnings in the windows build (hundreds).